src/linear_algebra/symplectic_group.lean

```diff
@@ -0,0 +1,245 @@
+/-
+Copyright (c) 2022 Matej Penciak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Matej Penciak, Moritz Doll, Fabien Clery
+-/
+
+import data.real.basic
+import linear_algebra.matrix.nonsingular_inverse
+
+/-!
+# The Symplectic Group
+
+This file defines the symplectic group and proves elementary properties.
+
+## Main Definitions
+
+`matrix.J`: the canonical `2n × 2n` skew-symmetric matrix
+`symplectic_group`: the group of symplectic matrices
+
+## Implementation Notes
+
+## TODO
+* Every symplectic matrix has determinant 1.
+* For `n = 1` the symplectic group coincides with the special linear group.
+-/
+
+open_locale matrix
+
+variables {l R : Type*}
+
+lemma pm_one_unit [ring R] {x : R} (h : x = 1 ∨ x = -1) : is_unit x :=
+begin
+  cases h,
+  { simp only [h, is_unit_one] },
+  { use -1,
+    simp only [h, units.coe_neg_one] }
+end
```

Comment:
I think this should be moved into `algebra/group/units`

Reply:
But honestly, I am not sure whether we want to have this lemma at all. We could also use [is_unit_iff_exists_inv](https://leanprover-community.github.io/mathlib_docs/algebra/group/units.html#is_unit_iff_exists_inv) and together with the fact that the square of the determinant is 1.

Reply:
I don't think we want this lemma. I think the neg case is just `is_unit_one.neg`